features/cellular/framework/targets/QUECTEL/BG96/QUECTEL_BG96_CellularStack.cpp

```diff
@@ -61,25 +85,49 @@ nsapi_error_t QUECTEL_BG96_CellularStack::socket_connect(nsapi_socket_t handle,
 
     _at.lock();
     if (socket->proto == NSAPI_TCP) {
-        char ipdot[NSAPI_IP_SIZE];
-        ip2dot(address, ipdot);
-        _at.at_cmd_discard("+QIOPEN", "=", "%d%d%s%s%d%d%d", _cid, request_connect_id, "TCP",
-                           ipdot, address.get_port(), socket->localAddress.get_port(), 0);
-
-        handle_open_socket_response(modem_connect_id, err);
-
-        if ((_at.get_last_error() == NSAPI_ERROR_OK) && err) {
-            if (err == BG96_SOCKET_BIND_FAIL) {
-                socket->id = -1;
+        if (socket->tls_socket) {
```

Comment:
What is the value of this when using MbedTLS?

Reply:
It is set as false for mbedtls based TLSSocket because in that case socket looks like a normal TCP socket from CellularStack point of view.